### PR TITLE
encoding: add front matter module

### PR DESCRIPTION
**Pull request by littledivy**
Closes https://github.com/denoland/deno_std/issues/2333

### CODE REVIEW

**Review by kt3k — APPROVED** [review]
LGTM

**Comment by kt3k** [comment]
BTW I'm also an author of this PR.

So please someone take another look on this PR?

**Review by bartlomieju — APPROVED** [review]

File: encoding/front_matter.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+ * @param str String to extract from.
+ *
+ * ```ts
+ * import { extract } from "https://deno.land/std@$STD_VERSION/encoding/front_matter.ts";
+ * import { assertEquals } from "https://deno.land/std@$STD_VERSION/testing/asserts.ts";
+ *
+ * const { attrs, body, frontMatter } = extract<{ title: string }>("---\ntitle: Three dashes marks the spot\n---\n");
+ * assertEquals(attrs.title, "Three dashes marks the spot");
+ * assertEquals(body, "");
+ * assertEquals(frontMatter, "---\ntitle: Three dashes marks the spot\n---\n");
```
Comment (geoffreypetri) [inline encoding/front_matter.ts:34]: Hello, the test here should be
`assertEquals(frontMatter, "title: Three dashes marks the spot");`
without the front matter payload.

Comment (kt3k) [reply encoding/front_matter.ts:34]: Good catch. Thanks!

**Pull request merged**